app/models/ext_management_system.rb

```diff
@@ -426,6 +426,24 @@ def enable!
     update!(:enabled => true)
   end
 
+  # Wait until all associated workers are dead to destroy this ems
+  def orchestrate_destroy
+    disable! if enabled?
+
+    workers = MiqWorker.find_current_or_starting.where(:queue_name => "ems_#{id}") #better way to get 'queue_name'?
```

Comment:
@jrafanie Heh, then again, can you remove the `unless` all together?
Just have both implement `queue_name`

(but please merge and don't end up in a debate over this)